java-spiffe-core/src/main/java/io/spiffe/workloadapi/retry/ExponentialBackoffPolicy.java

```diff
@@ -0,0 +1,82 @@
+package io.spiffe.workloadapi.retry;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Setter;
+import lombok.val;
+
+import java.time.Duration;
+import java.util.function.UnaryOperator;
+
+/**
+ * Represents a backoff policy for performing retries using exponential increasing delays.
+ */
+@Data
+public class ExponentialBackoffPolicy {
+
+    public static final ExponentialBackoffPolicy DEFAULT = new ExponentialBackoffPolicy();
+
+    // Retry indefinitely, default behavior.
+    public static final int UNLIMITED_RETRIES = 0;
+
+    private static final int BACKOFF_MULTIPLIER = 2;
+
+    // The first backoff delay period.
+    @Setter(AccessLevel.NONE)
+    private Duration initialDelay = Duration.ofSeconds(1);
+
+    // Max time of delay for the backoff period.
+    @Setter(AccessLevel.NONE)
+    private Duration maxDelay = Duration.ofSeconds(60);
+
+    // Max number of retries, unlimited by default.
+    @Setter(AccessLevel.NONE)
+    private int maxRetries = UNLIMITED_RETRIES;
+
+    @Setter(AccessLevel.NONE)
+    private int backoffMultiplier = BACKOFF_MULTIPLIER;
+
+    @Setter(AccessLevel.NONE)
+    private UnaryOperator<Duration> backoffFunction = d -> d.multipliedBy(backoffMultiplier);
```

Comment:
This backoff function is prone to integer overflow if the connection to the SPIFFE Workload API fails for an extended period of time.

I think we should probably compare the current delay to `maxDelay` before deciding to perform the multiplication, and only multiply iff the current delay is less than `maxDelay`.

Reply:
You are right, it would keep calculating an increasing new delay on each retry until reaching an overflow. Added a validation to prevent that situation.